src/ess/reduce/nexus/workflow.py

```diff
@@ -509,110 +552,58 @@ def _add_variances(da: sc.DataArray) -> sc.DataArray:
-            load_nexus_detector,
-            load_nexus_detector_data,
-            load_nexus_source,
-            load_nexus_sample,
-            get_source_position,
-            get_sample_position,
-            get_calibrated_detector,
-            assemble_detector_data,
-        )
-    )
-    wf[AnyRunPulseSelection] = AnyRunPulseSelection(slice(None, None))
+    """Generic workflow for loading detector data from a NeXus file."""
+    wf = sciline.Pipeline((*_common_providers, *_detector_providers))
     wf[DetectorBankSizes] = DetectorBankSizes({})
-    wf[AnyRunDetectorPositionOffset] = AnyRunDetectorPositionOffset(no_offset)
+    wf[PreopenNeXusFile] = PreopenNeXusFile(False)
     return wf
 
 
-def LoadNeXusWorkflow(filename: AnyRunFilename) -> sciline.Pipeline:
-    """
-    Workflow for loading detector and monitor data from a NeXus file.
-
-    On creation (as opposed to when the workflow is computed), the workflow will open
-    the file and determine the available detectors and monitors. This information is
-    used to create a workflow that loads all available detectors and monitors. This
-    can be useful when working with unknown files.
-
-    Parameters
-    ----------
-    filename:
-        NeXus file to load.
-    """
-    import pandas as pd
-
-    wf = sciline.Pipeline()
-    wf[AnyRunDetectorData] = LoadDetectorWorkflow()
-    wf[AnyRunAnyMonitorData] = LoadMonitorWorkflow()
-    wf[AnyRunFilename] = filename
-    wf.insert(nexus.read_nexus_file_info)
-    wf[nexus.NeXusFileInfo] = info = wf.compute(nexus.NeXusFileInfo)
-    # Note: There is a good reason against auto-mapping here:
-    # It will make extending the workflow much harder, as mapping should generally be
-    # done after the workflow is complete. We may consider splitting this.
-    # Note: What should we do when detector or monitor is incomplete? Parts of the
-    # workflow may make sense, but as a whole it will not run.
-    dets = [name for name, det in info.detectors.items() if det.n_pixel is not None]
-    det_df = pd.DataFrame({NeXusDetectorName: dets}, index=dets).rename_axis('detector')
-    mons = list(info.monitors)
-    mon_df = pd.DataFrame({AnyNeXusMonitorName: mons}, index=mons).rename_axis(
-        'monitor'
-    )
-    return wf.map(det_df).map(mon_df)
-
-
-def with_chunks(wf: sciline.Pipeline, chunk_length: sc.Variable) -> sciline.Pipeline:
-    """
-    Experimental helper to transform a workflow for chunked event-data processing.
-
-    Parameters
-    ----------
-    wf:
-        Workflow to transform.
-    chunk_length:
-        Length of chunks as a time difference, i.e., with a time unit.
-    """
-    import pandas as pd
-
-    info = wf.compute(nexus.NeXusFileInfo)
-    # start_time and end_time are taken from event_time_zero, so we always want to
-    # include the end
-    bounds = sc.arange(
-        'chunk', info.start_time, info.end_time + chunk_length, chunk_length
+def GenericNeXusWorkflow() -> sciline.Pipeline:
```

Comment:
```suggestion
def LoadNeXusWorkflow() -> sciline.Pipeline:
```
Since you removed 'generic' everywhere else.

Reply:
"Generic" here means "cross-technique".

Reply:
You didn't use that word for the other workflows in this file. And they are also cross-technique

Reply:
I don't see the point in changing naming in this context. If you think it is really important we can sit down and have a session on overall naming conventions.

Reply:
I don't think it is very important. I just think that 'GenericNeXusWorkflow' sounds like it does something conceptually different than 'LoadMonitorWorkflow'. Where actually, it just loads data and is as generic as 'LoadMonitorWorkflow'.